src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int GROW = 2;
+    private T[] arrayOfElements;
+    private int maxSize = 10;
+    private int size = 0;
+
+    public ArrayList() {
+        this.arrayOfElements = (T[]) new Object[maxSize];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == maxSize) {
+            grow();
+        }
+        arrayOfElements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheckForAdd(index);
+        if (size == maxSize) {
+            grow();
+        }
+        System.arraycopy(arrayOfElements, index, arrayOfElements, index + 1, size - index);
+        arrayOfElements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            if (size == maxSize) {
+                grow();
+            }
+            this.arrayOfElements[size] = list.get(i);
+            size++;
+        }
```

Comment:
Here it will be enough to use the already existing method
```suggestion
        for (int i = 0; i < list.size(); i++) {
              add(list.get(i));
        }
```